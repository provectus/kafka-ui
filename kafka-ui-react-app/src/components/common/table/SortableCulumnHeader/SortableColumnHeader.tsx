/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';
import cx from 'classnames';

import * as S from 'components/common/table/TableHeaderCell/TableHeaderCell.styled';

export interface ListHeaderProps {
  value: any;
  title: string;
  orderBy: any;
  setOrderBy: React.Dispatch<React.SetStateAction<any>>;
}

const ListHeaderCell: React.FC<ListHeaderProps> = ({
  value,
  title,
  orderBy,
  setOrderBy,
}) => (
  <th
    className={cx('is-clickable', orderBy === value && 'has-text-link-dark')}
    onClick={() => setOrderBy(value)}
  >
    {title}
    <S.SortIcon />
  </th>
);

export default ListHeaderCell;
