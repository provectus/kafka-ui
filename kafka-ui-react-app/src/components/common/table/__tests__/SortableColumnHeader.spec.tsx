import SortableColumnHeader from 'components/common/table/SortableCulumnHeader/SortableColumnHeader';
import { TopicColumnsToSort } from 'generated-sources';
import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

describe('ListHeader', () => {
  const setOrderBy = jest.fn();
  const component = (
    <table>
      <thead>
        <tr>
          <SortableColumnHeader
            value={TopicColumnsToSort.NAME}
            title="Name"
            orderBy={null}
            setOrderBy={setOrderBy}
          />
        </tr>
      </thead>
    </table>
  );

  describe('on column click', () => {
    it('calls setOrderBy', () => {
      render(component);
      userEvent.click(screen.getByRole('columnheader'));
      expect(setOrderBy).toHaveBeenCalledTimes(1);
      expect(setOrderBy).toHaveBeenCalledWith(TopicColumnsToSort.NAME);
    });
  });
});
