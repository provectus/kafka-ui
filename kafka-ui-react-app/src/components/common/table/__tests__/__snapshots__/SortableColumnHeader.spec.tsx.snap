// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListHeader matches the snapshot 1`] = `
<table>
  <thead>
    <tr>
      <ListHeaderCell
        orderBy={null}
        setOrderBy={[MockFunction]}
        title="Name"
        value="NAME"
      >
        <th
          className="is-clickable"
          onClick={[Function]}
        >
          Name
          <span
            className="icon is-small"
          >
            <i
              className="fas fa-sort"
            />
          </span>
        </th>
      </ListHeaderCell>
    </tr>
  </thead>
</table>
`;

exports[`ListHeader on column click matches the snapshot 1`] = `
<table>
  <thead>
    <tr>
      <ListHeaderCell
        orderBy={null}
        setOrderBy={[MockFunction]}
        title="Name"
        value="NAME"
      >
        <th
          className="is-clickable"
          onClick={[Function]}
        >
          Name
          <span
            className="icon is-small"
          >
            <i
              className="fas fa-sort"
            />
          </span>
        </th>
      </ListHeaderCell>
    </tr>
  </thead>
</table>
`;
